chiselFrontend/src/main/scala/chisel3/Data.scala

```diff
@@ -219,34 +219,32 @@ object chiselTypeOf {
 *
 * Thus, an error will be thrown if these are used on bound Data
 */
-object Input {
-  def apply[T<:Data](source: T)(implicit compileOptions: CompileOptions): T = {
+
+sealed abstract class BareDirection {
```

Comment:
I'll note that this is adding a new type to the type hierarchy which is an API addition. Since you're really just trying to collapse redundant code into a reusable function, my suggestion would be to add this as a `private[chisel3]` function to `object SpecifiedDirection` above, then they can call the function and it won't introduce a new type to the type hierarchy.